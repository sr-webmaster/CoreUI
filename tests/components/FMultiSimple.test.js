import { shallowMount, mount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import Component from '@freshinup/core-ui/src/components/FMultiSimple.vue'

export const FIXTURE_TAGS = [
  { uuid: '32423', name: 'Tag 1' },
  { uuid: '930183', name: 'Tag 2' }
]

describe('FMultiSimple', () => {
  // Component instance "under test"
  let localVue, mock
  describe('Snapshots', () => {
    beforeEach(() => {
      const vue = createLocalVue()
      localVue = vue.localVue
      mock = vue.mock
      mock
        .onGet('foodfleet/event-tags').reply(200, { data: FIXTURE_TAGS })
        .onAny().reply(config => {
          console.warn('No mock match for ' + config.url, config)
          return [404, {}]
        })
    })

    afterEach(() => {
      mock.restore()
    })

    test('defaults', () => {
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          url: 'foodfleet/event-tags',
          termParam: 'filter[name]',
          resultsIdKey: 'uuid'
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      const vue = createLocalVue()
      localVue = vue.localVue
      mock = vue.mock
      mock
        .onGet('foodfleet/event-tags').reply(200, { data: FIXTURE_TAGS })
        .onAny().reply(config => {
          console.warn('No mock match for ' + config.url, config)
          return [404, {}]
        })
    })

    afterEach(() => {
      mock.restore()
    })

    test('toggleMenu function chanage showMenu', () => {
      const wrapper = shallowMount(Component, {
        localVue,
        propsData: {
          url: 'foodfleet/event-tags',
          termParam: 'filter[name]',
          resultsIdKey: 'uuid'
        }
      })
      wrapper.setData({ showMenu: false })
      wrapper.vm.toggleMenu()
      expect(wrapper.vm.showMenu).toBeTruthy()
    })
  })

  describe('Computed', () => {
    beforeEach(() => {
      const vue = createLocalVue()
      localVue = vue.localVue
      mock = vue.mock
      mock
        .onGet('foodfleet/event-tags').reply(200, { data: FIXTURE_TAGS })
        .onAny().reply(config => {
          console.warn('No mock match for ' + config.url, config)
          return [404, {}]
        })
    })

    afterEach(() => {
      mock.restore()
    })

    test('showText', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          url: 'foodfleet/event-tags',
          termParam: 'filter[name]',
          resultsIdKey: 'uuid'
        }
      })
      wrapper.setData({ value: null })
      expect(wrapper.vm.showText).toBeNull()
      wrapper.setData({ value: [] })
      expect(wrapper.vm.showText).toBeNull()
      wrapper.setData({ value: [ { uuid: 1, name: 'mock' } ] })
      expect(wrapper.vm.showText).toBe('mock')
      wrapper.setData({ value: [ { uuid: 1, name: 'mock' }, { uuid: 2, name: 'mock' } ] })
      expect(wrapper.vm.showText).toBe('2 selected')
    })
  })
})
