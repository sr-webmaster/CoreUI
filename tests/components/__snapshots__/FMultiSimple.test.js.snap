// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FMultiSimple Snapshots defaults 1`] = `
<div
  class="v-menu"
  resultsidkey="uuid"
  termparam="filter[name]"
  url="foodfleet/event-tags"
>
  <div
    class="v-input v-text-field v-input--hide-details v-input--is-readonly theme--light"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <input
            readonly="readonly"
            resultsidkey="uuid"
            termparam="filter[name]"
            type="text"
            url="foodfleet/event-tags"
          />
        </div>
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--append"
          >
            <i
              aria-hidden="true"
              class="v-icon fas fa-caret-down theme--light"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="v-menu__content theme--light "
    style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
  >
     
    <div
      autofocus="autofocus"
      class="v-input v-text-field v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
      menu-props="[object Object]"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-select__slot"
          >
            <div
              class="v-select__selections"
            >
              <input
                aria-label=""
                autocomplete="off"
                autofocus="autofocus"
                placeholder="Start typing to Search"
                role="combobox"
                type="text"
              />
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--"
              >
                <i
                  aria-hidden="true"
                  class="v-icon v-icon--link material-icons theme--light"
                />
              </div>
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                >
                  arrow_drop_down
                </i>
              </div>
            </div>
          </div>
          <div
            class="v-menu"
          >
            <div
              class="v-menu__content theme--light v-autocomplete__content"
              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
            >
              <div
                class="v-select-list v-card theme--light"
              >
                <div
                  class="v-list theme--light"
                  role="list"
                >
                  <div
                    role="listitem"
                    style="display: none;"
                  >
                    <div
                      class="v-list__tile theme--light"
                    >
                      <span
                        class="search-section-title"
                      >
                        SELECTION
                      </span>
                    </div>
                  </div>
                    
                  <div
                    role="listitem"
                    style="display: none;"
                  >
                    <div
                      class="v-list__tile theme--light"
                    >
                      <span
                        class="search-section-title"
                      >
                        RESULTS
                      </span>
                    </div>
                  </div>
                  <div
                    role="listitem"
                  >
                    <div
                      class="v-list__tile theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          No data available
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
